### PR TITLE
Closes #247: bcva crit flag update

### LINKED ISSUE #247: Supersede `derive_var_bcvacritxfl()` and update vignettes/templates that use it
### Background Information

With the advent of [admiral::derive_vars_crit_flag()](https://pharmaverse.github.io/admiral/dev/reference/derive_vars_crit_flag.html) there is less of a need for `admiralophtha::derive_var_bcvacrtixfl()`. The latter should be superseded, in line with the {admiral} family's policy to not make many breaking changes post 1.0 releases.

### Definition of Done

- [x] `derive_var_bcvacritxfl()` superseded. See [here](https://github.com/pharmaverse/admiral/blob/main/R/derive_var_dthcaus.R) for an example of a superseded function from {admiral}.
- [x] templates/vignettesalso updated to not use it anymore and use `derive_vars_crit_flag()`, possibly in conjunction with `call_derivation()`.

### PR DESCRIPTION
**Pull request by steventing12**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update).
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files.
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions]
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately.
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralophtha (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralophtha site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralophtha/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`.
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`.
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@steventing12 please could you clean up this PR so it only includes the crterion flag updates for #247 and not the updates for #244 ? Otherwise it's hard to review. Thanks

File: inst/templates/ad_adbcva.R [hunk 1]
```diff
@@ -280,19 +280,25 @@ adbcva_adsl <- adbcva_aseq %>%
...
-      bcva_ranges = list(c(0, 5), c(-5, -1), c(10, 15)),
-      bcva_uplims = list(-20, 5, 10),
-      bcva_lowlims = list(-15, 15),
-      additional_text = ""
-    ),
-    filter = PARAMCD %in% c("SBCVA", "FBCVA")
-  ) %>%
+
+# Add criterion flags for BCVA endpoints
+adbcva_crtflag <- call_derivation(
```
Comment (manciniedoardo) [inline inst/templates/ad_adbcva.R:285]: Please could you move this section before the ASEQ derivation? Even though it makes no difference to the final output I think it makes logical sense to derive ASEQ as the final variable.

File: vignettes/adbcva.Rmd [hunk 2]
```diff
@@ -268,33 +268,42 @@ dataset_vignette(
...
 
-* Class 1: `CHG` value lying inside a range, `a <= CHG <= b`.
-* Class 2: `CHG` value below an upper limit, `CHG <= a`.
-* Class 3: `CHG` value above a lower limit, `CHG => b`.
+* `CHG` value lying inside a range, `CHG >= a & CHG <= b`
+* `CHG` value below an upper limit, `CHG <= a`
+* `CHG` value above a lower limit, `CHG => b`
 
-By using `derive_var_bcvacritxfl()`, the ADaM programmer can implement all three types of endpoint at once. This is achieved by feeding the appropriate ranges, upper limits and lower limits to the `bcva_ranges`, `bcva_uplims` and `bcva_lowlims` arguments of the function. For instance, let's suppose that the endpoints of interest are:
+and their corresponding description of the criterion. The critical flag `CRITxFL` is generated if `values_yn` is set to TRUE.
```
Comment (manciniedoardo) [inline vignettes/adbcva.Rmd:277]:
```suggestion
and their corresponding description of the criterion. The criterion flag `CRITxFL` is generated if `values_yn` is set to TRUE.
```
Second sentence needs a bit more clarity. For reference, the description of the argument `values_yn` reads: 
_Should "Y" and "N" be used for CRITyFL?If set to TRUE, the CRITyFL variable is set to "Y" if the condition (condition) evaluates to TRUE, it is set to "N" if the condition evaluate to FALSE, and to NA if it evaluates to NA.Otherwise, the CRITyFL variable is set to "Y" if the condition (condition) evaluates to TRUE, and to NA otherwise._

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks for the work @steventing12 - looks like a great start!

Along with the other comments, please could you also go ahead and supersede `derive_vars_bcvacritxfl()` - see [here](https://github.com/pharmaverse/admiral/blob/d7547a31095a708a7b0c06ed1c17addd4d39870f/R/derive_param_extreme_record.R#L132) for an example and please try render the website to ensure the lifecycle:superseded badge renders properly.

**Review by manciniedoardo — APPROVED** [review]
